third_party/amd/backend/compiler.py

```diff
@@ -331,7 +334,15 @@ def make_llir(src, metadata, options):
         # from memory.
         amd.set_all_fn_arg_inreg(fns[0])
 
-        if options.extern_libs:
+        if os.environ.get("TRITON_ENABLE_ADDRESS_SANITIZER", "0") == "1":
+            default_libdir = Path(__file__).parent / 'lib'
+            paths = [
+                str(default_libdir / 'asanrtl.bc'),
+                str(default_libdir / "ocml.bc"),
+                str(default_libdir / "ockl.bc")
+            ]
+            llvm.link_extern_libs(llvm_mod, paths)
+        elif options.extern_libs:
```

Comment:
When sanitizer is enabled, do we always need oc* libs?
Also, you can move this `llvm.link_extern_libs(llvm_mod, paths)` out of the if-else block

Reply:
Moved link_extern_libs. I'm not sure I have a definitive answer on if those other bc files always need to be included but I feel like including them is safer until we find a case where having them linked in causes problems.